data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/util/Joiner.java

```diff
@@ -105,12 +105,6 @@ private void joinAssociation(Path<?> path) {
             }
         } else if (path instanceof PersistentPropertyPath persistentPropertyPath) {
             Path parentPath = persistentPropertyPath.getParentPath();
-            if (parentPath instanceof PersistentAssociationPath parent) {
```

Comment:
This is the cause for the issue, we skip join where id was in criteria and then error is thrown that prop is unknown since relation is not joined.

Reply:
I think the check is valid for the associations that are storing the ID in the same table, in that case we don’t need to join.